crates/matrix-sdk-appservice/src/lib.rs

```diff
@@ -29,6 +29,16 @@
 //! The crate is in an experimental state. Follow
 //! [matrix-org/matrix-rust-sdk#228] for progress.
 //!
+//! The crate relies on the registration being always in sync with the actual
```

Comment:
I'd put it into its own "Registration" section, as this likely will keep being an requirement regardless of the crate status.